backend/routes/users.js

```diff
@@ -0,0 +1,77 @@
+const express = require('express');
+const bcrypt = require('bcrypt');
+const { PrismaClient } = require('@prisma/client');
+const prisma = new PrismaClient();
+
+const router = express.Router();
+
+router.post('/', async (req, res) => {
+    const { username, email, password } = req.body;
+
+    try {
+        const existingUser = await prisma.user.findFirst({
+            where: {
+                OR: [
+                    { username: username },
+                    { email: email }
+                ]
+            }
+        });
+
+        if (existingUser) {
+            return res.status(400).json({ error: 'Username or email already exists' });
+        }
+
+        const hashedPassword = await bcrypt.hash(password, 10);
```

Comment:
seems like this library handles generating the salt?

normally i would have expected:
1. some secure method to generate salt for hashing function
2. hashing using the provided password + salt
3. storing both the hash and the salt